### PR TITLE
Demonstrate datetime example with static data

### PR DESCRIPTION
**Pull request by sffc**
This should not be broken, but it is broken:

> thread '<unnamed>' panicked at 'Failed to create DateTimeFormat instance.: DataProvider(Resource("Serde Deserialization Error"))', components/datetime/examples/work_log.rs:60:10
> note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
> fatal runtime error: failed to initiate panic, error 5
> Aborted

StaticDataProvider seems to work with all other formatters, but not DateTimeFormat?

### CODE REVIEW

**Comment by sffc** [comment]
Minimal repro checked in as *de.rs*.  Reproduce with:

```
$ cd components/datetime
$ cargo test de
```

**Comment by sffc** [comment]
If this fails, I suspect that round-tripping `DatePatternsV1` through Postcard also fails.

File: components/datetime/tests/de.rs [hunk 1]
```diff
@@ -0,0 +1,8 @@
+use icu_datetime::provider::*;
+
+#[test]
+fn de() {
```
Comment (zbraniecki) [inline components/datetime/tests/de.rs:4]: `suggestion`: can you pick a more descriptive name for the test? It shows up in test list and should be possible to filter for with `cargo test X`.

**Review by zbraniecki — APPROVED** [review]

**Comment by sffc** [comment]
No, basic roundtrip seems to work.  Will investigate more later.

File: components/datetime/tests/de.rs (hunk 1 above)
Comment (sffc) [reply components/datetime/tests/de.rs:4]: I don't intend to check this in; I opened this draft PR to demonstrate an issue I'm trying to debug that I didn't know if you'd have any thoughts on.

**Comment by sffc** [comment]
I found the problem.  Closing in favor of #947

**Pull request closed without merging**